pkg/controller/rhmiconfig/rhmiconfig_controller.go

```diff
@@ -113,6 +114,28 @@ func (r *ReconcileRHMIConfig) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
+	//	Checks if there is an upgrade available, if there is upgrade available upgrades the status of RHMIConfig
+	if rhmiConfig.Status.UpgradeAvailable != nil {
+		if reschedule, ok := rhmiConfig.Annotations[integreatlyv1alpha1.RecalculateScheduleAnnotation]; ok && reschedule == "true" {
+			err = helpers.UpdateStatus(context.TODO(), r.client, rhmiConfig)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+
+			rhmiConfig.Annotations[integreatlyv1alpha1.RecalculateScheduleAnnotation] = ""
+			err = r.client.Update(context.TODO(), rhmiConfig)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+	} else {
+		rhmiConfig.Status.Upgrade = integreatlyv1alpha1.RHMIConfigStatusUpgrade{}
+		err = r.client.Update(context.TODO(), rhmiConfig)
```

Comment:
```suggestion
		rhmiConfig.Status.Upgrade = integreatlyv1alpha1.RHMIConfigStatusUpgrade{
		    Scheduled: nil,
		}
		err = r.client.Status().Update(context.TODO(), rhmiConfig)
```

Mainly need to call `.Status()`, otherwise it won't be updated

Reply:
:+1: